src/ioctls/vm.rs

```diff
@@ -418,36 +430,61 @@ impl VmFd {
+    /// // ASM code that just forces a MMIO Write.
+    /// let asm_code = [
+    ///         0xc6, 0x06, 0x00, 0x80, 0x00,
+    /// ];
+    /// #[cfg(target_arch = "aarch64")]
+    /// let asm_code = [
+    ///     0x01, 0x00, 0x00, 0x10, /* adr x1, <this address> */
+    ///     0x22, 0x10, 0x00, 0xb9, /* str w2, [x1, #16]; write to this page */
+    ///     0x02, 0x00, 0x00, 0xb9, /* str w2, [x0]; force MMIO exit */
+    ///     0x00, 0x00, 0x00, 0x14, /* b <this address>; shouldn't get here, but if so loop forever */
     /// ];
     ///
     /// // Write the code in the guest memory. This will generate a dirty page.
     /// unsafe {
     ///     let mut slice = slice::from_raw_parts_mut(load_addr, mem_size);
-    ///     slice.write(&x86_code).unwrap();
+    ///     slice.write(&asm_code).unwrap();
     /// }
     ///
     /// let vcpu_fd = vm.create_vcpu(0).unwrap();
     ///
-    /// let mut vcpu_sregs = vcpu_fd.get_sregs().unwrap();
-    /// vcpu_sregs.cs.base = 0;
-    /// vcpu_sregs.cs.selector = 0;
-    /// vcpu_fd.set_sregs(&vcpu_sregs).unwrap();
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// {
+    ///     // x86_64 specific registry setup.
+    ///     let mut vcpu_sregs = vcpu_fd.get_sregs().unwrap();
+    ///     vcpu_sregs.cs.base = 0;
+    ///     vcpu_sregs.cs.selector = 0;
+    ///     vcpu_fd.set_sregs(&vcpu_sregs).unwrap();
+    ///
+    ///     let mut vcpu_regs = vcpu_fd.get_regs().unwrap();
+    ///     // Set the Instruction Pointer to the guest address where we loaded the code.
+    ///     vcpu_regs.rip = guest_addr;
+    ///     vcpu_regs.rax = 2;
+    ///     vcpu_regs.rbx = 3;
+    ///     vcpu_regs.rflags = 2;
+    ///     vcpu_fd.set_regs(&vcpu_regs).unwrap();
+    /// }
     ///
-    /// let mut vcpu_regs = vcpu_fd.get_regs().unwrap();
-    ///  // Set the Instruction Pointer to the guest address where we loaded the code.
-    /// vcpu_regs.rip = guest_addr;
-    /// vcpu_regs.rax = 2;
-    /// vcpu_regs.rbx = 3;
-    /// vcpu_regs.rflags = 2;
-    /// vcpu_fd.set_regs(&vcpu_regs).unwrap();
+    /// #[cfg(target_arch = "aarch64")]
+    /// {
+    ///     // aarch64 specific registry setup.
+    ///     let mut kvi = kvm_bindings::kvm_vcpu_init::default();
+    ///     vm.get_preferred_target(&mut kvi).unwrap();
+    ///     vcpu_fd.vcpu_init(&kvi).unwrap();
+    ///
+    ///     let core_reg_base: u64 = 0x6030_0000_0010_0000;
+    ///     let mmio_addr: u64 = guest_addr + mem_size as u64;
+    ///     vcpu_fd.set_one_reg(core_reg_base + 2 * 32, guest_addr); // set PC
+    ///     vcpu_fd.set_one_reg(core_reg_base + 2 * 0, mmio_addr); // set X0
+    /// }
     ///
     /// loop {
     ///     match vcpu_fd.run().expect("run failed") {
-    ///         VcpuExit::Hlt => {
-    ///             // The code snippet dirties 1 page when loading the code in memory.
+    ///         VcpuExit::MmioWrite(addr, data) => {
+    ///             // The code snippet dirties its own page.
```

Comment:
On x86_64 loading the code dirties one page. On aarch64 it looks like you need a write to the guest memory. The comment needs to be updated as well.

Reply:
I updated it, take a look.